src/scales/scale.time.js

```diff
@@ -562,6 +562,37 @@ module.exports = function(Chart) {
 			me._minorFormat = formats[unit];
 			me._majorFormat = formats[majorUnit];
 			me._table = buildLookupTable(me._timestamps.data, min, max, options.distribution);
+			me._leftOffset = 0;
+			me._rightOffset = 0;
+
+			if (options.offset && ticks.length) {
```

Comment:
BTW, did you try to compute offset based on `me._timestamps.data` instead of `ticks`?